pkg/docker/image.go

```diff
@@ -101,3 +103,23 @@ func FindImage(ctx context.Context, ow *rpc.OutputWriter, client *client.Client,
 	}
 	return nil, false, nil
 }
+
+func GetImageID(ctx context.Context, cli *client.Client, defaultTag string) (string, error) {
+	filters := filters.NewArgs()
+	filters.Add("reference", defaultTag)
+	listOpts := types.ImageListOptions{
+		Filters: filters,
+	}
+
+	images, err := cli.ImageList(ctx, listOpts)
+	if err != nil {
+		return "", fmt.Errorf("docker image list failed: %w", err)
+	}
+
+	if len(images) != 1 {
+		return "", fmt.Errorf("unexpected number of images returned by docker image list, expected 1, got: %d", len(images))
+	}
+
+	// get 3cde7451eb28 from sha256:3cde7451eb28a3199f2c7d4e8e02a98f2e96b9a34dd4a9bc7eeaa5a192a1536f
+	return images[0].ID[7 : 7+12], nil
```

Comment:
There's an implicit precondition here that we're not testing against. We should do a `HasPrefix` assertion and panic if false.

Reply:
I'll push this change and merge.